lib/src/klio/transforms/decorators.py

```diff
@@ -309,42 +310,101 @@ def func_wrapper(*args, **kwargs):
     return func_wrapper
 
 
-def _handle_klio(func_or_meth):
-    @functools.wraps(func_or_meth)
-    def method_wrapper(self, *args, **kwargs):
-        with _klio_context() as ctx:
-            setattr(self, "_klio", ctx)
-
-        # SO. HACKY. We check to see if this method is named "expand"
-        # to designate  if the class is a Composite-type transform
-        # (rather than a DoFn with a "process" method).
-        # A Composite transform handles a pcoll / pipeline,
-        # not the individual elements, and therefore doesn't need
-        # to be given a KlioMessage. It should only need the KlioContext
-        # attached.
-        if func_or_meth.__name__ == "expand":
-            return func_or_meth(self, *args, **kwargs)
-
-        wrapper = __serialize_klio_message
-        # Only the process method of a DoFn is a generator - otherwise
-        # beam can't pickle a generator
-        if __is_dofn_process_method(self, func_or_meth):
-            wrapper = __serialize_klio_message_generator
+def __get_thread_limiter(max_thread_count, thread_limiter):
+    # max_thread_count=None or True will default to CPU count
```

Comment:
I find it a little confusing to mix boolean values into something that's supposed to be an integer - could we do something instead using enums or default constants like, 

```
#max_thread_count=None will turn off thread limits
#max_thread_count=ThreadLimits.DEFAULT will set it to default CPU count
```